src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -505,6 +509,60 @@ class UserService(
       case None => IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"User $workbenchUserId not found")))
     }
   }
+
+  def countDirectSynchronizedGroupMemberships(samUser: SamUser, samRequestContext: SamRequestContext): IO[Int] =
+    directoryDAO.countDirectSynchronizedGroupMemberships(samUser, samRequestContext)
+
+  def countIndirectSynchronizedGroupMemberships(samUser: SamUser, samRequestContext: SamRequestContext): IO[Int] =
+    directoryDAO.countIndirectSynchronizedGroupMemberships(samUser, samRequestContext)
+
+  def countUnsynchronizedGroupMemberships(samUser: SamUser, samRequestContext: SamRequestContext): IO[Int] =
+    directoryDAO.countUnsynchronizedGroupMemberships(samUser, samRequestContext)
+
+  def getSamUserCombinedState(
+      workbenchEmail: WorkbenchEmail,
+      samRequestContext: SamRequestContext,
+      resourceService: ResourceService
+  ): IO[SamUserCombinedStateResponse] =
+    for {
+      samUserOption <- getUserFromEmail(workbenchEmail, samRequestContext)
+      samUser = samUserOption.getOrElse(throw new WorkbenchException("user not found"))
+      combinedState <- getSamUserCombinedState(samUser, samRequestContext, resourceService)
+    } yield combinedState
+
+  def getSamUserCombinedState(samUser: SamUser, samRequestContext: SamRequestContext, resourceService: ResourceService): IO[SamUserCombinedStateResponse] =
+    for {
+      allowances <- getUserAllowances(samUser, samRequestContext)
+      maybeAttributes <- getUserAttributes(samUser.id, samRequestContext)
+      directGroupMemberships <- countDirectSynchronizedGroupMemberships(samUser, samRequestContext)
+      indirectGroupMemberships <- countIndirectSynchronizedGroupMemberships(samUser, samRequestContext)
+      unsynchronizedGroupMemberships <- countUnsynchronizedGroupMemberships(samUser, samRequestContext)
+      termsOfServiceDetails <- tosService.getTermsOfServiceDetailsForUser(samUser.id, samRequestContext)
+      enterpriseFeatures <- resourceService
+        .listResourcesFlat(
+          samUser.id,
+          Set(ResourceTypeName("enterprise-feature")),
+          Set.empty,
+          Set(ResourceRoleName("user")),
+          Set.empty,
+          includePublic = false,
+          samRequestContext
+        )
+      favoriteResources <- resourceService.getUserFavoriteResources(samUser.id, samRequestContext)
+    } yield SamUserCombinedStateResponse(
+      samUser,
+      allowances,
+      maybeAttributes,
+      termsOfServiceDetails.getOrElse(TermsOfServiceDetails(None, None, permitsSystemUsage = false, isCurrentVersion = false)),
+      GroupMembershipCounts(
+        directSynchronized = directGroupMemberships,
+        totalSynchronized = indirectGroupMemberships,
+        unsynchronized = unsynchronizedGroupMemberships
+      ),
+      Map("enterpriseFeatures" -> enterpriseFeatures.toJson),
+      favoriteResources
+    )
+
```

Comment:
I think it's kinda smelly to require `ResourceService` as an argument to this method. But I also didn't want to update all 67 calls to the UserService constructor just to support this one method.

Reply:
I guess I agree. An alternative would be to create another Service class that depends on both but 🤷 . I leave it up to you.